### DISCUSSION TITLE
[Solved] Roborock S7 not showing Web UI

### DISCUSSION THREAD

**Discussion by fjosegalan** [Hypfer/Valetudo#1874] [Q&A / Support] [answered]
Hello all,

I finally decided to root my Roborock S7 after running for a year with Xiaomi integration with Home assistant.
I rooted the device using the FEL method, and follow the main guidelines from Valetudo web.
1.- Initial Root with laptop attached to usb and short GND-TPA17 was OK using the FEL firmware from Dustbuilder. Reboot
2.- Backed up NANDs was OK, I have it stored in my laptop.
3.- First firmware installation from Dustbuilder was OK. Reboot.
4.- Second firmware installation from Dustbuilder was OK. reboot
5.- Upload Valetudo upx firmware to /mnt/data/valetudo was OK.Reboot.
6.- Reboot and the Wifi AP is enabled.

From this point on, I cannot do nothing else....

AP wifi point is shown and I can connect to it, in fact connected to it I can ssh and scp in to my S7 without issues. But I cannot reach http://192.168.8.1 valetudo web ui to configure nothing. I did all the steps twice and nothing changed, it still the same.
As a hint I can state that first time installing FEL firmware from laptop I am not really sure if I waited enough time for the reboot.... consequently I am not sure if some files could be potentially damaged, second attempt I did not dismantled the speakers and I am pretty sure the device had been rebooted before continue with next step due it had been confirmed by the sounds.
To summarize, I can SSH and SCP to my device, but WebUI is not shown and It is in a "ghost" state due to cannot configure wifi settings in new firmware and old parameters are no longer valid and does not connect back to wifi. The device still working as usual if you manage it from the buttons (Cleans as usually)

Could you give some guidelines to continue my troubleshooting? I am running out of ideas...
Can I check if Valetudo and needed services are running? Systemctl and service commands are not working and I am not sure how to check it.
Can I restore my nand backups back to the device again? I did not find any info about that.
If further information is needed, please ask. I want to use Valetudo and fix it.

Regards

Fran Galan

**Comment by Hypfer** [comment, marked as answer]
Usually the issue is that people take this command from the install guide
```
scp -O -i ~/.ssh/your_keyfile Downloads/valetudo-armv7-lowmem.upx root@192.168.8.1:/mnt/data/valetudo
```
and instead execute
```
scp -O -i ~/.ssh/your_keyfile Downloads/valetudo-armv7-lowmem.upx root@192.168.8.1:/mnt/data/valetudo/
```
which is wrong as `/mnt/data/valetudo` should be the binary itself.


Since you said you can SSH into the machine, it should already be rooted, meaning that the issue is likely a valetudo binary in the wrong place or a valetudo binary that is actually just an html file

**Reply by fjosegalan** [reply]
Hello Hypfer

For sure is like that.... I created manually the valetudo folder inside /mnt/data/ and then move the upx maintaining the same name valetudo-armv7-lowmem.upx using WinSCP due to my Desktop Linux PC (Used for rooting) has no wifi card, and my laptop is runing on Windows. 
For this, What I understand I should do:
1.- Remove /mnt/data/valetudo/ folder (and any contents)
2.- Move valetudo-armv7-lowmem.upx to /mnt/data/
3.- Rename valetudo-armv7-lowmem.upx to valetudo?
I am little bit confussed due to upx file is about 10.6MB, binary file is 27.5MB... on Github
Please clarify me which file I need to put on /mnt/data/, wich name I need to rename and if I need to keep the upx extension or not.

Thanks in advance Hypfer, Great job!!!!

Fran Galan

**Comment by fjosegalan** [comment]
Hello

Issue is fixed as per Hypfer comments.
Valetudo binary was wrongly copied to folder /mnt/data/valetudo/. As Hypfer commented the Valetudo binary should be copied to /mnt/data/ with the name valetudo with any extension.
Also don’t forget to change the execution permision of the binary file:
`chmod +x /mnt/data/valetudo`
